source/iface/device-selector.rst

```diff
@@ -3,116 +3,94 @@
   SPDX-License-Identifier: CC-BY-4.0
 
 .. _device-selectors:
+.. _device_selector:
 
 ****************
 Device selectors
 ****************
 
-Devices selectors allow the SYCL runtime to choose the device.
+Device selectors allow the SYCL runtime to choose the device.
 
 A device selector can be passed to :ref:`queue`, :ref:`platform`, and
 other constructors to control the selection of a device. A program may
-use `Built-in Device Selectors`_ or define its own device_selector_
-for full control.
-
-.. _device_selector:
-
-.. rst-class:: api-class
-
-=========================
-``sycl::device_selector``
-=========================
-
-::
-
-  class device_selector;
-
-Abstract class for device selectors.
-
-This is the base class for the `Built-in Device Selectors`_. To define
-a custom device selector, create a derived class that defines the ()
-operator.
-
-.. todo:: Example Custom device selector
-
-(constructors)
-==============
-
-::
-
-  device_selector();
-  device_selector(const sycl::device_selector &rhs);
-
-Construct a device_selector.
-
-A device selector can be created from another by passing ``rhs``.
-
+use `Built-in Device Selectors`_ or define its own device_selector for
+full control.
 
-``select_device``
-=================
+Interface for a device selector is any object that meets the C++ named
+requirement ``Callable`` taking a const :ref:`device` reference and
+returning a value implicitly convertible to a int.
 
-::
+At any point where the SYCL runtime needs to select a SYCL device using
+a device selector, the system queries all root devices from all SYCL
+backends in the system, calls the device selector on each device and
+selects the one which returns the highest score. If the highest
+value is strictly negative no device is selected.
 
-  sycl::device select_device() const;
+In places where only one device has to be picked and the high score is
+obtained by more than one device, then one of the tied devices will be
+returned, but which one is not defined and may depend on enumeration
+order, for example, outside the control of the SYCL runtime.
 
 
-Returns the device with the highest score as determined by calling
-`operator()`_.
-
-.. rubric:: Exceptions
-
-Throws a runtime error if all devices have a negative score.
-
-``operator=``
-=============
-
-::
-
-  sycl::device_selector &operator=(const sycl::device_selector &rhs);
+.. _built-in-device-selectors:
 
-Create a device selector by copying another one.
+===========================
+Built-in Device Selectors
+===========================
 
+Standard device selectors included with all SYCL implementations:
 
-``operator()``
-==============
+.. list-table::
 
-::
+  * - ``default_selector_v``
+    - Select a SYCL device from any supported SYCL backend based on an implementation-defined
+      heuristic. Must select the host device if no other suitable device can be found.
```

Comment:
Remove this sentence:

> Must select the host device if no other suitable device can be found.

There is no such thing as the "host device" in SYCL 2020.

I think you should add a sentence like:

> Since all implementations must support at least one device, this selector must always return a device.

Reply:
I agree, it sounds better, thanks for the recommendation!
I added it in the latest commit

Reply:
I agree, it sounds better, thanks for the recommendation!
I added it in the latest commit